syntax = "proto3";

message ClientControl {
  message StartStream {
    string foo = 1;
  }
  message StopStream {
    string foo = 1;
  }
  oneof command {
    StartStream start_stream = 1;
    StopStream stop_stream = 2;
  }
}

message ClientData {
  message Hello {
    string hostname = 1;
    string username = 2;
  }
  message Data {
    bytes payload = 1;
  }

  oneof type {
    Hello hello = 1;
    Data data = 2;
  };
}

